app/services/trainees/create_from_apply.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+module Trainees
+  class CreateFromApply
+    include ServicePattern
+
+    def initialize(application:)
+      @application = application
+    end
+
+    def call
+      trainee.save!
+      trainee
+    end
+
+  private
+
+    attr_reader :application
+
+    def attributes
+      @attributes ||= parsed_application["attributes"]
+    end
+
+    def parsed_application
+      @parsed_application ||= JSON.parse(application.application)
+    end
+
+    def trainee
+      @trainee ||= Trainee.new(mapped_attributes)
+    end
+
+    def mapped_attributes
+      {
+        apply_application: application,
+        provider: application.provider,
+        trainee_id: raw_trainee["id"],
+        first_names: raw_trainee["first_name"],
+        last_name: raw_trainee["last_name"],
+        date_of_birth: raw_trainee["date_of_birth"],
+        email: raw_contact_details["email"],
+        # TODO: Use actual route type, if found.
```

Comment:
Left over from the spike and hardcoded currently - we need to resolve. How do we know what training route they're actually on?

Reply:
We should be able to cross reference from the courses table when we get that going properly.

Reply:
Got it - pulling from the course in our table, if present.